spec/lib/open_food_network/order_and_distributor_report_spec.rb

```diff
@@ -1,48 +1,155 @@
+            let(:o2) { create(:order, distributor: d1, completed_at: 1.day.ago, bill_address: create(:address), ship_address: create(:address)) }
+            let(:li2) { build(:line_item, product: create(:simple_product, supplier: s1)) }
+
+            before do
+              o2.line_items << li2
+              create(:enterprise_relationship, parent: s1, child: d1, permissions_list: [:add_to_order_cycle])
+            end
 
-      it "should return a header row describing the report" do
-        subject = OrderAndDistributorReport.new [@order]
+            it "shows line items supplied by my producers, with names hidden" do
+              subject.table_items.should == [li2]
+              subject.table_items.first.order.bill_address.firstname.should == "HIDDEN"
+            end
+          end
 
-        header = subject.header
-        header.should == ["Order date", "Order Id",
-          "Customer Name","Customer Email", "Customer Phone", "Customer City",
-          "SKU", "Item name", "Variant", "Quantity", "Max Quantity", "Cost", "Shipping cost",
-          "Payment method",
-          "Distributor", "Distributor address", "Distributor city", "Distributor postcode", "Shipping instructions"]
+          context "that has not granted P-OC to the distributor" do
+            let(:o2) { create(:order, distributor: d1, completed_at: 1.day.ago, bill_address: create(:address), ship_address: create(:address)) }
+            let(:li2) { build(:line_item, product: create(:simple_product, supplier: s1)) }
+
+            before do
+              o2.line_items << li2
+            end
+
+            it "shows line items supplied by my producers, with names hidden" do
+              subject.table_items.should == []
+            end
+          end
+        end
+
+        context "as a manager of a distributor" do
+          let!(:user) { create(:user) }
+          subject { OrderAndDistributorReport.new user }
+
+          before do
+            d1.enterprise_roles.create!(user: user)
+          end
+
+          it "only shows line items distributed by enterprises managed by the current user" do
+            d2 = create(:distributor_enterprise)
+            d2.enterprise_roles.create!(user: create(:user))
+            o2 = create(:order, distributor: d2, completed_at: 1.day.ago)
+            o2.line_items << build(:line_item)
+            subject.table_items.should == [li1]
+          end
+
+          it "only shows the selected order cycle" do
+            oc2 = create(:simple_order_cycle)
+            o2 = create(:order, distributor: d1, order_cycle: oc2)
+            o2.line_items << build(:line_item)
+            subject.stub(:params).and_return(order_cycle_id_in: oc1.id)
+            subject.table_items.should == [li1]
+          end
+        end
       end
 
-      it "should denormalise order and distributor details for display as csv" do
-        subject = OrderAndDistributorReport.new [@order]
+      context "Old version" do
```

Comment:
How to you feel about describing these contexts as `html` and `json` instead of `old` and `new`?